internal/primitives/blockchain/backend.go

```diff
@@ -10,6 +10,37 @@ import (
 	"github.com/ChainSafe/gossamer/internal/primitives/runtime/generic"
 )
 
+// BlockBackend is an interface for fetching block data
+type BlockBackend[H runtime.Hash, N runtime.Number, Header runtime.Header[N, H],
```

Comment:
nit: can we expand the generic params to new lines like existing code in `internal/client`? (ex. https://github.com/ChainSafe/gossamer/blob/9489a08cf0f5b2b7c3986f51a4acab5e0fb08e45/internal/client/api/backend.go#L40)